Error "java.lang.IllegalStateException: Could not read child element: parent" is generated  for a maven module

### Issue
The following error is generated when we compile a maven module where there is no maven pom parent

### Error
```
Caused by: java.lang.IllegalStateException: Could not read child element: parent
    at io.ap4k.project.MavenInfo.lambda$getElement$6 (MavenInfo.java:145)
    at java.util.Optional.orElseThrow (Optional.java:290)
    at io.ap4k.project.MavenInfo.getElement (MavenInfo.java:145)
    at io.ap4k.project.MavenInfo.getParentVersion (MavenInfo.java:109)
    at io.ap4k.project.MavenInfo.getVersion (MavenInfo.java:89)
    at io.ap4k.project.MavenInfo.<init> (MavenInfo.java:45)
    at io.ap4k.project.MavenInfoReader.getInfo (MavenInfoReader.java:45)
    at io.ap4k.project.MavenInfoReader.getInfo (MavenInfoReader.java:29)
    at io.ap4k.project.FileProjectFactory.lambda$getProjectInfo$2 (FileProjectFactory.java:78)
    at java.util.Optional.map (Optional.java:215)
    at io.ap4k.project.FileProjectFactory.getProjectInfo (FileProjectFactory.java:78)
    at io.ap4k.project.FileProjectFactory.createInternal (FileProjectFactory.java:59)
    at io.ap4k.project.FileProjectFactory.create (FileProjectFactory.java:48)
    at io.ap4k.project.AptProjectFactory.createInternal (AptProjectFactory.java:52)
    at io.ap4k.project.AptProjectFactory.create (AptProjectFactory.java:42)
    at io.ap4k.processor.AbstractAnnotationProcessor.init (AbstractAnnotationProcessor.java:54)
```

--- a/core/src/main/java/io/ap4k/Session.java
+++ b/core/src/main/java/io/ap4k/Session.java
@@ -36,7 +36,7 @@ import java.util.function.Consumer;
  */
 public class Session {
 
-  private static Session INASTANCE;
+  private static Session INSTANCE;
 
   private final AtomicBoolean closed = new AtomicBoolean();
   private final Set<Generator> generators = new LinkedHashSet<>();
@@ -48,15 +48,15 @@ public class Session {
    * @return  The Session.
    */
   public static Session getSession() {
-    if (INASTANCE != null) {
-      return INASTANCE;
+    if (INSTANCE != null) {
+      return INSTANCE;
     }
     synchronized (Session.class) {
-      if (INASTANCE == null) {
-        INASTANCE = new Session();
+      if (INSTANCE == null) {
+        INSTANCE = new Session();
       }
     }
-    return INASTANCE;
+    return INSTANCE;
   }
 
   public Configurations configurations() {

--- a/core/src/main/java/io/ap4k/project/MavenInfo.java
+++ b/core/src/main/java/io/ap4k/project/MavenInfo.java
@@ -17,6 +17,7 @@
  **/
 package io.ap4k.project;
 
+import io.ap4k.utils.Strings;
 import org.w3c.dom.Document;
 import org.w3c.dom.Element;
 import org.w3c.dom.Node;
@@ -80,18 +81,17 @@ public class MavenInfo implements BuildInfo {
 
     /**
      * Read the version form the document.
-     * @pairam document     The document.
+     * @param document     The document.
      * @return              The version, if exists, the parent version otherwise.
      */
     public static String getVersion(Document document) {
-        return getElement(document.getDocumentElement(), VERSION)
-            .map(e -> e.getTextContent())
-            .orElse(getParentVersion(document));
+        String version = getElement(document.getDocumentElement(), VERSION).map(e -> e.getTextContent()).orElse(null);
+        return Strings.isNotNullOrEmpty(version) ? version : getParentVersion(document);
     }
 
     /**
      * Read the packaging form the document.
-     * @pairam document     The document.
+     * @param document     The document.
      * @return              The version, if exists, the parent version otherwise.
      */
     public static String getPackaging(Document document) {

